src/RuleSet/DeclarationBlock.php

```diff
@@ -21,14 +27,33 @@
  * matching elements.
  *
  * Declaration blocks usually appear directly inside a `Document` or another `CSSList` (mostly a `MediaQuery`).
+ *
+ * Note that `CSSListItem` extends both `Commentable` and `Renderable`, so those interfaces must also be implemented.
  */
-class DeclarationBlock extends RuleSet
+class DeclarationBlock implements CSSElement, CSSListItem, Positionable, RuleContainer
 {
+    use CommentContainer;
+    use Position;
+
     /**
      * @var array<Selector|string>
      */
     private $selectors = [];
 
+    /**
+     * @var RuleSet
+     */
+    private $ruleSet;
+
+    /**
+     * @param int<1, max>|null $lineNumber
+     */
+    public function __construct(?int $lineNumber = null)
+    {
+        $this->ruleSet = new RuleSet($lineNumber);
```

Comment:
We also need unit tests that the created `RuleSet` uses has line number from the `DeclarationBlock` constructor, and that it defaults to `null`.

Reply:
We don't appear to have an existing test that `RuleSet` applies the line number provided, so perhaps another pre-PR is needed...

Reply:
#1348 for `RuleSet`.  Once that is approved, I can address this along similar lines.

Reply:
Have added tests to cover:
1. `getLineNumber()` returning line number passed to `DeclarationBlock` constructor (or `null` by default);
2. `getRuleSet()->getLineNumber()` returning line number passed to `DeclarationBlock` constructor (or `null` by default);

But perhaps `1` could be covered by another pre-PR...